src/objects.h

```diff
@@ -841,7 +841,13 @@ EXPORT_INLINE Obj TYPE_COMOBJ(Obj obj)
 */
 EXPORT_INLINE void SET_TYPE_COMOBJ(Obj obj, Obj val)
 {
+#ifdef HPCGAP
+    MEMBAR_WRITE();
+    // Only require a read guard
+    ((Obj *) CONST_ADDR_OBJ(obj))[0] = val;
```

Comment:
A minor problem here is that `UNSAFE_ADDR_OBJ()` is generated by unward and not available otherwise. This currently needs an additional `#ifdef USE_HPC_GUARDS`, or we can use `UNSAFE_PTR_BAG()` instead, which returns the same value. But yes, I can definitely change that.

Reply:
I am fine with using UNSAFE_PTR_BAG